mongo/description/server.go

```diff
@@ -459,7 +459,34 @@ func (s Server) Equal(other Server) bool {
 		return false
 	}
 
-	// the rest of the server fields are not used in sdam monitoring server descriptions and not compared
+	if s.LastError != other.LastError {
```

Comment:
This isn't a safe way to compare errors. Errors generated via mechanisms like `errors.New` won't be equal even if they have the same message. This can come up in unexpected ways because some of these errors come from external sources like Go's `net` package. See https://play.golang.org/p/jwAe6j1mH1l for an example.